src/bout++.cxx

```diff
@@ -193,6 +193,9 @@ int BoutInitialise(int &argc, char **&argv) {
     }
   }
   bool color_output = false; // Will be set true if -c is in the options
+  char ** argvcleaned = (char**) malloc(sizeof(char *)*(argc+1));
```

Comment:
Please don't use `malloc`! This should just be a `std::vector`

Reply:
I wasn't sure as other parts of the code might assume it is a plain pointer, and thus call free or remalloc - which I though is not safe if the memory has not been allocated directly by malloc.

Reply:
I think that's undefined behaviour.

This might be cleaner if you use a `std::vector<std::string>` to store the original `argv` for later when we write it out, and then just set `argv[i-1][0] = argv[i][0] = 0;` (to set the flag and its argument to empty string).